types/euwa-wrapper/index.d.ts

```diff
@@ -0,0 +1,213 @@
+     */
+    hooks?: Hooks;
+}
+
+export interface ApplicationList {
+    [app: string]: string;
+}
+
+export interface Hooks {
+    /**
+     * Only publish/subscribe interface is available
+     * @param event
+     * @param callback
+     * @returns function
+     */
+    onBeforeLoad: () => void;
+    [hook: string]: () => void;
+}
+
+export interface ApplicationBridge {
+    /**
+     * Returns object with minimal chat state
+     * Similar to clicking the start button
+     * It will open the chat window and show the next view
+     * @returns ApplicationAPI
+     */
+    api: ApplicationAPI;
+
+    /**
+     * Publish custom events, use this to build custom functions to do other work
+     * @example
+     * import {EUWA} from '@puzzel/euwa-wrapper';
+     * const euwa = new EUWA({
+     *    configId: 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
+     *    customerKey: 123456
+     *   }, {
+     *   hooks: {
+     *       // All hooks accept functions
+     *      onBeforeLoad: publishMyCustomEvent
+     *    }
+     * });
+     *
+     * function publishMyCustomEvent() {
+     *    const chat = euwa.getApplicationBeforeLoad(EUWA.APPS.CHAT);
+     *    chat.publish('PublishEvent', data => {
+     *        console.log('PublishEvent Data:', data);
+     *    });
+     * }
+     * @param event
+     * @param data
+     * @returns
+     */
+    publish: (event: string, ...data: any) => void;
+
+    /**
+     * Subscribe to custom event
+     * @param event
+     * @param callback
+     *
+     * @example
+     * import {EUWA} from '@puzzel/euwa-wrapper';
+     *
+     *  const euwa = new EUWA({
+     *       configId: 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
+     *       customerKey: 123456
+     *  }, {
+     *       hooks: {
+     *            // All hooks accept functions
+     *             onBeforeLoad: subscribeToChatInit
+     *          }
+     *     }
+     * );
+     *
+     * function subscribeToChatInit() {
+     *    // Get the Chat's event interface
+     *    const chat = euwa.getApplicationBeforeLoad(EUWA.APPS.CHAT);
+     *
+     *    // Subscribe to chatInit* event
+     *    chat.subscribe('chatInit', data => {
+     *        console.log('Chat Init Data:', data);
+     *    });
+     * }
+     */
+    subscribe: (event: string, callback: () => void) => void;
+}
+
+export interface ApplicationSettings {
+    [app: string]: object;
+}
+
+export interface SystemVariables {
+    enteredFormName: string;
+    enteredChatId: string;
+    enteredFormIssue: string;
+    selectedQueueKey: string;
+    timeId2Map: string;
+}
+
+export interface ChatState {
+    isConnected: boolean;
+    isEnded: boolean;
+    isMinimized: boolean;
+}
+
+export interface ApplicationAPI {
+    /**
+     * Returns object with minimal chat state
+     * Similar to clicking the start button
+     * It will open the chat window and show the next view
+     * @returns ChatState
+     */
+    getState: () => ChatState;
```

Comment:
Are these supposed to be represented as fields or would methods make more sense?

Reply:
Hi,

Following their example [here](https://help.puzzel.com/product-documents/user-guide/puzzel-contact-centre/puzzel-administration-portal/services/chat-configuration/euwa-wrapper-interface) I kept it as a method

```ts
declare interface ApplicationAPI {
    [method: string]: Function
}
```

I created this for our own use on sats.no in our typescript/react site, and thought it would be nice to share it with one interested.

thanks for the review, my first time doing this, all tips & suggestions welcome :-)